assembly/src/parsers/crypto_ops.rs

```diff
@@ -137,8 +137,16 @@ fn mtree_get(span_ops: &mut Vec<Operation>) {
     span_ops.push_many(Operation::Dup7, 4);
 
     // move d, i back to the top of the stack => [d, i, V, R, V, ...]
-    span_ops.push(Operation::MovUp13);
-    span_ops.push(Operation::MovUp13);
+    // MovUp13
+    span_ops.push(Operation::SwapDW);
+    span_ops.push(Operation::MovUp5);
+    span_ops.push(Operation::SwapDW);
+    span_ops.push(Operation::MovUp8);
+    // MovUp13
+    span_ops.push(Operation::SwapDW);
+    span_ops.push(Operation::MovUp5);
+    span_ops.push(Operation::SwapDW);
+    span_ops.push(Operation::MovUp8);
```

Comment:
I haven't spent much time thinking about it, but I'm wondering if there is anything we can do with the overall logic to make the whole thing a bit more efficient.

Reply:
I managed to optimize the sequential call of two operations `MovUp9` and `MovDn9` --- now each pair requires 6 cycles instead of 8, however, I have not found a way to optimize the sequential call of two operations `MovUp13`. An alternative option that at least works requires as many as 10 cycles. If we could use the `MovUp9` and `MovDn9` operations, then this number could be reduced to 6 (and, obviously, to two for `MovUp9` and `MovDn9`), but it seems that then the whole point of reducing the number of native operations is lost.
But there is a high probability that I just overlooked the option of optimizing it.

Reply:
By going through all possible combinations of operations on the stack using BFS, it turned out that it was impossible to achieve the desired result faster than in 8 cycles.

Reply:
What about if look at things before this sequence. I haven't thought this through myself - but maybe something like this will work:

The stack starts in the state `[d, i, R, ...]` and then we do two `MovDn5`'s to get it to be `[R, d, i, ...]`. Maybe instead we do something like:
```
pad
pad
sawpw
```
The state now will be `[R, 0, 0, d, i, ...]`. But then maybe at the end instead of using moving individual elements back to the top, we could do something like:
```
swapw3
drop
drop
```
And the overall effect would be the same?

Reply:
Indeed! Concentrating on the task of optimizing two `MovUp13` operations, I completely forgot that it is possible to use the previous operations too. Using them, we can achieve an increase in cycles by only 1, compared to what it was before the introduction of `MovUp8` and other operations.

Reply:
great! Could something similar be applied to `prep_stack_for_mrupdate()`? Or it doesn't work there?